src/net.cpp

```diff
@@ -2530,7 +2530,13 @@ void CConnman::AddNewAddresses(const std::vector<CAddress>& vAddr, const CAddres
 
 std::vector<CAddress> CConnman::GetAddresses()
 {
-    return addrman.GetAddr();
+    std::vector<CAddress> addresses = addrman.GetAddr();
+    if (m_banman) {
+        addresses.erase(std::remove_if(addresses.begin(), addresses.end(),
```

Comment:
nit: I think that's a slight perfomance regression as you will iterate twice on the non-erased vector elements. Likely not to matter as it's not a hot point.

Reply:
No. remove-erase removes unwanted elements in a single pass.

Reply:
Previously, in net_processing, we iterate on `vAddr` and filter out discouraged/banned elements before to push them in `vAddrToSend`. So that's a single pass.

Now, in net.cpp we iterate on `addresses` on every element, then in net_processing we iterate again on `vAddr` to push  elements in `vAddrToSend`. So that's two pass on the valid set ?

Reply:
Ah, ok. Yes, you're right that for peers with "addr" permissions, we'll iterate through the list twice (although the list will be max size 1000 instead of 2500, so still fewer items). However, in the common case, we'll only iterate through the cached list once, and not have to call IsDiscouraged() or IsBanned() on that list.